docs/src/DevDocs/Common/Thermodynamics/saturation_adjustment.jl

```diff
@@ -0,0 +1,213 @@
+            zlabel = "Total specific humidity",
+            title = "3D input to PhaseEquil",
+            xlims = (min(ρ_all...), max(ρ_all...)),
+            ylims = energy_axis ? (min(e_int_all...), max(e_int_all...)) :
+                    (min(T_mask...), max(T_mask...)),
+            zlims = (min(q_tot_all...), max(q_tot_all...)),
+        )
+        savefig(joinpath(folder, filename))
+    end
+
+    # 2D binned scatter plots
+    function plot2D_slices(ts_no_err, ts, NM; energy_axis, converged)
+        mask = converged ? ts .≠ nothing : ts .== nothing
+        ρ_mask = ρ_all[mask]
+        if energy_axis
+            var = e_int_all[mask]
+        else
+            var = air_temperature.(ts_no_err[mask])
+        end
+        q_tot_mask = q_tot_all[mask]
+        c_name = converged ? "converged" : "non_converged"
+        label = converged ? "converged" : "non-converged"
+        short_name = converged ? "C" : "NC"
+        e_name = energy_axis ? "energy" : "temperature"
+        nm_name = nameof(NM)
+        filename = "2DSlice_$(e_name)_$(c_name)_$nm_name.svg"
+        filename = joinpath(folder, filename)
+        save_binned_surface_plots(
+            ρ_mask,
+            var,
+            q_tot_mask,
+            short_name,
+            filename;
+            xlims = (min(ρ_mask...), max(ρ_mask...)),
+            ylims = energy_axis ? (min(e_int_all...), max(e_int_all...)) :
+                    (min(var...), max(var...)),
+            label = label,
+            ref_points = prof_pts,
+        )
+    end
+
+    #! format: off
+    for NM in numerical_methods
+    # Energy axis
+    plot3D(ts_no_err[NM], ts[NM], NM; energy_axis = true, include_prof = true, converged = false);
+    plot3D(ts_no_err[NM], ts[NM], NM; energy_axis = true, include_prof = true, converged = true);
+
+    plot2D_slices(ts_no_err[NM], ts[NM], NM; energy_axis = true, converged=true);
+    plot2D_slices(ts_no_err[NM], ts[NM], NM; energy_axis = true, converged=false);
+
+
+    # Temperature axis
+    plot3D(ts_no_err[NM], ts[NM], NM; energy_axis = false, include_prof = false, converged = false);
+    plot3D(ts_no_err[NM], ts[NM], NM; energy_axis = false, include_prof = false, converged = true);
+    plot2D_slices(ts_no_err[NM], ts[NM], NM; energy_axis = false, converged=true);
+    plot2D_slices(ts_no_err[NM], ts[NM], NM; energy_axis = false, converged=false);
+    end
+    #! format: on
+
+    convergence_percent = Dict()
+    for NM in numerical_methods
+        convergence_percent[NM] = count(ts[NM] .≠ nothing) / length(ts[NM])
+    end
+    println("Convergence percentages:")
+    for (k, v) in convergence_percent
+        println("$k = $v")
+    end
+
+    @warn "Note that the temperature axis for the non-converged
+    plot is not necessarily accurate, since the temperatures are
+    the result of a non-converged saturation adjustment. Additionally,
+    non-converged cases (resulting in NaNs) do not appear in the
+    non-converged plots"
```

Comment:
Why don't you plot the true temperature instead? I think this would be more revealing w.r.t. where failure occurs.

Reply:
We can plot true temperature, I'll adjust the temperature plot to use true temperature, and I'll remove this warning. One reason I wanted to still have internal energy was so that, if we could ensure convergence in a known 3D input space, we could use something like
```julia
L1 = e_int_min < e_int < e_int_max
L2 = ρ_min < ρ < ρ_max
L3 = q_tot_min < q_tot < q_tot_max
if !all((L1,L2,L3))
    error("Attempting to initialize an unphsical
        thermodynamic state with e_int=$e_int, ρ=$ρ, q_tot=$q_tot")
end
```
in `PhaseEquil`. At this point in the code, we don't have temperature.

Reply:
I prefer not to have such hard bounds in the code. They have caused me much grief (and time wasted tracking them down in code) earlier in my career: What's unphysical on Earth is not necessarily unphysical on other planets, and it's nice to be able to simulate other planets.

Reply:
Ah, ok, that's fair. I'll change over to the true temperature plots entirely.